lib/llvm-backend/src/backend.rs

```diff
@@ -204,25 +218,140 @@ impl LLVMBackend {
+                } else {
+                    format!("fn{}", index)
+                };
+
+                let c_str = CString::new(name).unwrap();
+                let ptr = unsafe { get_func_symbol(module, c_str.as_ptr()) };
+
+                assert!(!ptr.is_null());
+                local_func_id_to_addr.push(ptr as usize);
+            }
+
+            let mut addr_to_size_record: BTreeMap<usize, &StkSizeRecord> = BTreeMap::new();
+
+            for record in &map.stk_size_records {
+                addr_to_size_record.insert(record.function_address as usize, record);
+            }
+
+            let mut map_records: BTreeMap<usize, &StkMapRecord> = BTreeMap::new();
+
+            for r in &map.stk_map_records {
+                map_records.insert(r.patchpoint_id as usize, r);
+            }
+
+            for ((start_id, start_entry), (end_id, end_entry)) in stackmaps
+                .entries
+                .iter()
+                .enumerate()
+                .step_by(2)
+                .zip(stackmaps.entries.iter().enumerate().skip(1).step_by(2))
+            {
+                if let Some(map_record) = map_records.get(&start_id) {
+                    assert_eq!(start_id, map_record.patchpoint_id as usize);
+                    assert!(start_entry.is_start);
+                    assert!(!end_entry.is_start);
+
+                    let end_record = map_records.get(&end_id);
+
+                    let addr = local_func_id_to_addr[start_entry.local_function_id];
+                    let size_record = *addr_to_size_record
+                        .get(&addr)
+                        .expect("size_record not found");
+
+                    start_entry.populate_msm(
+                        module_info,
+                        code_ptr as usize,
+                        &map,
+                        size_record,
+                        map_record,
+                        end_record.map(|x| (end_entry, *x)),
+                        &mut msm,
+                    );
+                } else {
+                    // TODO: optimized out?
+                }
+            }
+
+            let code_ptr = unsafe { llvm_backend_get_code_ptr(module) } as usize;
+            let code_len = unsafe { llvm_backend_get_code_size(module) } as usize;
+
+            let local_func_id_to_offset: Vec<usize> = local_func_id_to_addr
+                .iter()
+                .map(|&x| {
+                    assert!(x >= code_ptr && x < code_ptr + code_len);
+                    x - code_ptr
+                })
+                .collect();
+
+            //println!("MSM: {:?}", msm);
+
+            (
+                Self {
+                    module,
+                    buffer: Arc::clone(&buffer),
+                    msm: Some(msm),
+                    local_func_id_to_offset,
+                },
+                LLVMCache { buffer },
+            )
+        } else {
+            eprintln!("WARNING: No stack map");
```

Comment:
I  believe this warning could be removed. This prints multiple times in the spectests currently.